sqldeveloper: 17.4.1.054.0712 -> 18.2.0.183.1748

Ma27 opened the pull request:
###### Motivation for this change

Bump to the latest available version ATM:
http://www.oracle.com/technetwork/developer-tools/sql-developer/downloads/index.html

Additionally switch back to OracleJDK, it is officially supported and
far more stable when working with the software:
https://community.oracle.com/docs/DOC-888316#jive_content_id_22_About_OpenJDK_and_Oracle_JDK

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ma27 committed:
sqldeveloper: 17.4.1.054.0712 -> 18.2.0.183.1748

Bump to the latest available version ATM:
http://www.oracle.com/technetwork/developer-tools/sql-developer/downloads/index.html

Additionally switch back to OracleJDK, it is officially supported and
far more stable when working with the software:
https://community.oracle.com/docs/DOC-888316#jive_content_id_22_About_OpenJDK_and_Oracle_JDK

GrahamcOfBorg commented:
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44624&attempt_id=3e1b8192-3548-4178-a699-dbd0d60763b3)

The following builds were skipped because they don't evaluate on x86_64-linux: sqldeveloper

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44624&attempt_id=cc0e488a-43c4-4043-83bf-525d225f8530)

The following builds were skipped because they don't evaluate on aarch64-linux: sqldeveloper

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

flokli commented:
@Ma27 I looked at bumping to 18.2, but decided to defer this for now, as sqldeveloper started using JavaFX with 18.2. (so currently wont work with openjdk anymore at all).

Personally, I'd prefer to use `openjdk`, (and add `JavaFX` support via ` OpenJFX`,) to avoid having to manually download the oracle JDK, as license disallows manual downloading, and also given there's no oraclejdk support for darwin in nixpkgs either.

Ma27 commented:
you're right I guess, although the SQL developer is more stable with the Oracle SDK (according to Oracle^^). As I currently use and need SQL developer v18, I'll publish the changes in my personal overlay fore now. I'd leave this open (for now if that's okay) until SQL developer 17 is EOLed, then we can continue evaluate which JDK to use for this package.

flokli commented:
@Ma27 Na, no need - I can imagine there's usecases where people might need to use the 18.x version too, even having to manually download yet another binary is not that appealing.

I'd like to keep sqldeveloper usable for Darwin users, and given we don't support oraclejdk on that platform in nixpkgs, I'm not very keen on making 18.2 default for now, but keep it as a separate derivation.

I rebased your commit, splitted it separate `18.2.nix` file, exposed it in top-level as `sqldeveloper_18`, and added myself and you as maintainers ;-)
You can find the changes at https://github.com/flokli/nixpkgs/tree/sqldeveloper-18.2 . Let me know if this works for you :-)

Ma27 commented:
First of all, thank you for your effort regarding my change and the introduction of Darwin support for `sqldeveloper`!

> I can imagine there's usecases where people might need to use the 18.x version too, even having to manually download yet another binary is not that appealing.

Agreed! I tried to get SQL developer v18 work with OpenJDK, but didn't manage to because of several JavaFX issues which is why I decided to use OracleJDK for now, I know that this isn't optimal though :/

> I'd like to keep sqldeveloper usable for Darwin users, and given we don't support oraclejdk on that platform in nixpkgs, I'm not very keen on making 18.2 default for now, but keep it as a separate derivation.

I didn't know about this as I'm not an OS X user, but yes, it would be a wrong decision to drop Darwin support again.

> I rebased your commit, splitted it separate 18.2.nix file, exposed it in top-level as sqldeveloper_18, and added myself and you as maintainers ;-)
You can find the changes at https://github.com/flokli/nixpkgs/tree/sqldeveloper-18.2 . Let me know if this works for you :-)

I just checked your branch out and built it locally - works fine!
I'll close this now, feel free to ping me when opening a PR from your sqldeveloper branch against Nixpkgs master :)

flokli commented:
@Ma27 No worries :-)

I opened #45341 which adds sqldeveloper 18.2 to nixpkgs.
